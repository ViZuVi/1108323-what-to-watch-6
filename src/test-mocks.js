export const movies = [{
  id: 1,
  title: `Macbeth`,
  backgroundColor: `#F1E9CE`,
  backgroundImg: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Macbeth.jpg`,
  description: `Macbeth, the Thane of Glamis, receives a prophecy from a trio of witches that one day he will become King of Scotland. Consumed by ambition and spurred to action by his wife, Macbeth murders his king and takes the throne for himself.`,
  director: `Justin Kurzel`,
  genre: `Drama`,
  isavorite: false,
  posterImg: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Macbeth.jpg`,
  previewImg: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/macbeth.jpg`,
  previewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  rating: 3.3,
  released: 2015,
  runTime: 113,
  scores: 48798,
  starring: [`Michael Fassbender`, `Marion Cotillard`, `Jack Madigan`],
  videoLink: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
},
{
  id: 2,
  title: `Macbeth`,
  backgroundColor: `#F1E9CE`,
  backgroundImg: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Macbeth.jpg`,
  description: `Macbeth, the Thane of Glamis, receives a prophecy from a trio of witches that one day he will become King of Scotland. Consumed by ambition and spurred to action by his wife, Macbeth murders his king and takes the throne for himself.`,
  director: `Justin Kurzel`,
  genre: `Drama`,
  isavorite: false,
  posterImg: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Macbeth.jpg`,
  previewImg: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/macbeth.jpg`,
  previewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  rating: 3.3,
  released: 2015,
  runTime: 113,
  scores: 48798,
  starring: [`Michael Fassbender`, `Marion Cotillard`, `Jack Madigan`],
  videoLink: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
}];

export const movie = {
  id: 1,
  title: `Macbeth`,
  backgroundColor: `#F1E9CE`,
  backgroundImg: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Macbeth.jpg`,
  description: `Macbeth, the Thane of Glamis, receives a prophecy from a trio of witches that one day he will become King of Scotland. Consumed by ambition and spurred to action by his wife, Macbeth murders his king and takes the throne for himself.`,
  director: `Justin Kurzel`,
  genre: `Drama`,
  isavorite: false,
  posterImg: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Macbeth.jpg`,
  previewImg: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/macbeth.jpg`,
  previewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  rating: 3.3,
  released: 2015,
  runTime: 113,
  scores: 48798,
  starring: [`Michael Fassbender`, `Marion Cotillard`, `Jack Madigan`],
  videoLink: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
};

export const comments = [
  {
    id: 1,
    user: {
      id: 4,
      name: `Kate Muir`
    },
    rating: 8.9,
    comment: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
    date: `2019-05-08T14:13:56.569Z`
  },
  {
    id: 2,
    user: {
      id: 4,
      name: `Kate Muir`
    },
    rating: 8.9,
    comment: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
    date: `2019-05-08T14:13:56.569Z`
  },
];

export const genres = [`All genres`, `Drama`, `Comedy`];

export const userInfo = {
  id: 1,
  email: `Oliver.conner@gmail.com`,
  name: `Oliver.conner`,
  avatarUrl: `img/1.png`
};

