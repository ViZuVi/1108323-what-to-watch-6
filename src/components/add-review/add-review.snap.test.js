import React from 'react';
import {render} from '@testing-library/react';
import {Router} from 'react-router-dom';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import {createMemoryHistory} from 'history';
import AddReview from './add-review';
import {movie} from '../../test-mocks';

const mockStore = configureStore({});

it(`AddReview should render correctly`, () => {
  const history = createMemoryHistory();
  const store = mockStore({
    ACTIVE_MOVIE: {movie},
  });
  const {container} = render(
      <Provider store={store}>
        <Router history={history}>
          <AddReview />
        </Router>
      </Provider>
  );

  expect(container).toMatchSnapshot();
});
