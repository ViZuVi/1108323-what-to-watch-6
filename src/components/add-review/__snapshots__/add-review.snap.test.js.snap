// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddReview should render correctly 1`] = `
<div>
  <section
    class="movie-card movie-card--full"
  >
    <div
      class="movie-card__header"
    >
      <div
        class="movie-card__bg"
      >
        <img
          alt="Macbeth"
          src="https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Macbeth.jpg"
        />
      </div>
      <h1
        class="visually-hidden"
      >
        WTW
      </h1>
      <header
        class="page-header"
      >
        <div
          class="logo"
        >
          <a
            class="logo__link"
            href="/"
          >
            <span
              class="logo__letter logo__letter--1"
            >
              W
            </span>
            <span
              class="logo__letter logo__letter--2"
            >
              T
            </span>
            <span
              class="logo__letter logo__letter--3"
            >
              W
            </span>
          </a>
        </div>
        <nav
          class="breadcrumbs"
        >
          <ul
            class="breadcrumbs__list"
          >
            <li
              class="breadcrumbs__item"
            >
              <a
                class="breadcrumbs__link"
                href="movie-page.html"
              >
                Macbeth
              </a>
            </li>
            <li
              class="breadcrumbs__item"
            >
              <a
                class="breadcrumbs__link"
              >
                Add review
              </a>
            </li>
          </ul>
        </nav>
        <div
          class="user-block"
        >
          <div
            class="user-block__avatar"
          />
        </div>
      </header>
      <div
        class="movie-card__poster movie-card__poster--small"
      >
        <img
          alt="The Grand Budapest Hotel poster"
          height="327"
          src="https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Macbeth.jpg"
          width="218"
        />
      </div>
    </div>
    <div
      class="add-review"
    >
      <form
        action="#"
        class="add-review__form"
      >
        <div
          class="rating"
        >
          <div
            class="rating__stars"
          >
            <input
              class="rating__input"
              data-testid="stars"
              id="star-0"
              name="rating"
              type="radio"
              value="1"
            />
            <label
              class="rating__label"
              for="star-0"
            >
              Rating 
              0
            </label>
            <input
              class="rating__input"
              data-testid="stars"
              id="star-1"
              name="rating"
              type="radio"
              value="2"
            />
            <label
              class="rating__label"
              for="star-1"
            >
              Rating 
              1
            </label>
            <input
              class="rating__input"
              data-testid="stars"
              id="star-2"
              name="rating"
              type="radio"
              value="3"
            />
            <label
              class="rating__label"
              for="star-2"
            >
              Rating 
              2
            </label>
            <input
              class="rating__input"
              data-testid="stars"
              id="star-3"
              name="rating"
              type="radio"
              value="4"
            />
            <label
              class="rating__label"
              for="star-3"
            >
              Rating 
              3
            </label>
            <input
              class="rating__input"
              data-testid="stars"
              id="star-4"
              name="rating"
              type="radio"
              value="5"
            />
            <label
              class="rating__label"
              for="star-4"
            >
              Rating 
              4
            </label>
            <input
              class="rating__input"
              data-testid="stars"
              id="star-5"
              name="rating"
              type="radio"
              value="6"
            />
            <label
              class="rating__label"
              for="star-5"
            >
              Rating 
              5
            </label>
            <input
              class="rating__input"
              data-testid="stars"
              id="star-6"
              name="rating"
              type="radio"
              value="7"
            />
            <label
              class="rating__label"
              for="star-6"
            >
              Rating 
              6
            </label>
            <input
              class="rating__input"
              data-testid="stars"
              id="star-7"
              name="rating"
              type="radio"
              value="8"
            />
            <label
              class="rating__label"
              for="star-7"
            >
              Rating 
              7
            </label>
            <input
              class="rating__input"
              data-testid="stars"
              id="star-8"
              name="rating"
              type="radio"
              value="9"
            />
            <label
              class="rating__label"
              for="star-8"
            >
              Rating 
              8
            </label>
            <input
              class="rating__input"
              data-testid="stars"
              id="star-9"
              name="rating"
              type="radio"
              value="0"
            />
            <label
              class="rating__label"
              for="star-9"
            >
              Rating 
              9
            </label>
          </div>
        </div>
        <div
          class="add-review__text"
        >
          <textarea
            class="add-review__textarea"
            data-testid="comment"
            id="review-text"
            maxlength="400"
            minlength="50"
            name="review-text"
            placeholder="Review text"
          />
          <div
            class="add-review__submit"
          >
            <button
              class="add-review__btn"
              disabled=""
              type="submit"
            >
              Post
            </button>
          </div>
        </div>
      </form>
    </div>
  </section>
</div>
`;
